streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicManager.java

```diff
@@ -169,6 +173,9 @@ public void makeReady(final Map<String, InternalTopicConfig> topics) {
             log.error(timeoutAndRetryError);
             throw new StreamsException(timeoutAndRetryError);
         }
+        log.debug("Completed validating internal topics and created {}", newlyCreatedTopics);
```

Comment:
I think this race condition was particularly severe since we do the listOffsets request pretty much immediately after creating the topics, whereas whatever we're doing with that topic next will not be until the rebalance was completed.

AFAIK we've never had any users report subsequent operations failing after the first rebalance due to not-yet-fully-created topics, but it could have just slipped past us

Reply:
I do agree it would be useful though. Feel free to create a ticket :P

Reply:
Thanks! Will do. I just wanted to bounce the idea off you first, in case it was stupid.